lib/Dialect/FIRRTL/Transforms/IMConstProp.cpp

```diff
@@ -111,6 +111,18 @@ class LatticeValue {
     return getValue().dyn_cast_or_null<IntegerAttr>();
   }
 
+  void dump() const {
+    if (isInvalidValue())
+      llvm::errs() << "invalid";
+    if (isUnknown())
+      llvm::errs() << "unknown";
+    if (isOverdefined())
+      llvm::errs() << "overdefined";
+    if (isConstant()) {
+      llvm::errs() << "constant(" << getConstant().getValue() << ")";
+    }
+  }
```

Comment:
It would be more standard to define the `<<` operator for `LatticeValue`. (This also isn't used right now.  I'm completely in favor of adding `LLVM_DEBUG` code to IMCP in the future, though.)